### PR TITLE
Handle tuple deserialization in deserialize_any properly

### LINKED ISSUE #121: Ron does not properly round trip through Value
```rust
#[macro_use]
extern crate serde;

extern crate ron;
use ron::de::*;
use ron::ser;
use ron::value::Value;

#[derive(Serialize)]
struct Scene(Option<(u32, u32)>);

#[derive(Serialize)]
struct Scene2 {
    foo: Option<(u32, u32)>,
}

fn main() {
    {
        let s = ser::to_string(&Scene2 { foo: Some((1, 1)) }).unwrap();
        println!("{}", s);
        let scene: Value = from_str(&s).unwrap();
        println!("{:?}", scene);
    }
    {
        let s = ser::to_string(&Scene( Some((1, 1)) )).unwrap();
        println!("{}", s);
        let scene: Value = from_str(&s).unwrap();
        println!("{:?}", scene);
    }
}
```

The first example gives:
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Parser(ExpectedIdentifier, Position { col: 12, line: 1 })', libcore/result.rs:945:5
```

The second gives:
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: Parser(ExpectedMapColon, Position { col: 6, line: 1 })', libcore/result.rs:945:5
```

### PR DESCRIPTION
**Pull request by torkleyy**
Fixes #121 

Needs some more tests before merging. I'm not happy about all the lookahead that this requires, but I don't see a better solution.

### CODE REVIEW

**Comment by torkleyy** [comment]
Okay, let's see if bors works now.

bors try

**Comment by torkleyy** [comment]
Okay, now only one issue remains: The conversion from a serialized tuple to a `Value` loses the tuple information (it becomes a sequence). That's because Serde's data model doesn't differentiate between those two.

File: src/de/mod.rs [hunk 1]
```diff
@@ -92,6 +92,32 @@ impl<'de> Deserializer<'de> {
...
+    /// Called from `deserialze_any` when a struct was detected. Decides if there is a
+    /// unit, tuple or usual struct and deserializes it accordingly.
+    ///
+    /// This method assumes there is no identifier left.
+    fn handle_any_struct<V>(&mut self, visitor: V) -> Result<V::Value>
+    where
+        V: Visitor<'de>
+    {
+        // Create a working copy
+        let mut bytes = self.bytes.clone();
```
Comment (kvark) [inline src/de/mod.rs]: this method doesn't mutate the actual byte stream even on success?

Comment (torkleyy) [reply src/de/mod.rs]: It does, but through calling `self.deserialize_tuple` / `self.deserialize_struct`. I need to create a working copy for lookahead first, the bytes do get consumed later.

**Comment by torkleyy** [comment]
Thank you for the review!
bors r+

File: Cargo.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [package]
 name = "ron"
-version = "0.3.0"
+version = "0.4.0"
```
Comment (matklad) [inline Cargo.toml]: Would you like to make a release? I am hitting this exact issue right now :rofl:

**Comment by torkleyy** [comment]
Sure!

**Comment by torkleyy** [comment]
Done ;)

**Pull request merged**